File: code/sorting/gnome_sort/gnome_sort.java
// Java Program to implement Gnome Sort
 
import java.util.Arrays;
public class GnomeSort
{
    static void gnomeSort(int arr[], int n)
    {
        int index = 0;
 
        while (index < n)
        {
            if (index == 0)
                index++;
            if (arr[index] >= arr[index-1])
                index++;
            else
            {
                int temp =0;
                temp = arr[index];
                arr[index] = arr[index-1];
                arr[index-1] = temp;
                index--;
            }
        }
        return;
    }
 
    // Driver program to test above functions.
    public static void main(String[] args)
    {
        int arr[] = {34, 2, 10, -9};
 
        gnomeSort(arr, arr.length);
 
        System.out.print("Sorted sequence after applying Gnome sort: ");
        System.out.println(Arrays.toString(arr));
    }
}

File: code/sorting/src/gnome_sort/gnome_sort.js
function gnomeSort(arr) {
  let index = 0;

  while (index <= arr.length) {
    if (index > 0 && arr[index - 1] > arr[index]) {
      let tmp = arr[index];
      arr[index] = arr[index - 1];
      arr[index - 1] = tmp;
      --index;
    } else {
      --index;
    }
  }
}

// test case
const a = [3, 6, 2, 9, 8, 1, 4, 0, 3, 7, 7];
console.log(gnomeSort(a));
